components/admin_console/custom_plugin_settings/index.ts

```diff
@@ -17,24 +19,26 @@ import {it} from '../admin_definition';
 
 import {appsPluginID} from 'utils/apps';
 
-import CustomPluginSettings from './custom_plugin_settings.jsx';
+import CustomPluginSettings from './custom_plugin_settings';
 import getEnablePluginSetting from './enable_plugin_setting';
 
+type OwnProps = { match: { params: { plugin_id: string } } }
+
 function makeGetPluginSchema() {
     return createSelector(
         'makeGetPluginSchema',
-        (state, pluginId) => state.entities.admin.plugins[pluginId],
-        (state, pluginId) => getAdminConsoleCustomComponents(state, pluginId),
+        (state: GlobalState, pluginId: string) => state.entities.admin.plugins![pluginId],
```

Comment:
this might run into issues when plugins is not set in production ... I would prefer trying to get that value and if not present return an empty object, or something that would prevent the selector to break.

I am not 100% familiar with how this part of the store is built, so I would ask @hmhealey for advice.

Reply:
Waiting for feedback for this change

Reply:
Unfortunately this doesn't prevent Typescript complain for `plugins`: `Object is possibly 'undefined'.` Therefore I believe I still need `!` after `plugins`. As you mentioned, there is already null check in the beginning, keeping current code should create any issue.

Reply:
Could you try to use the optional chaining instead? If we check for `null` anyways this could do the trick!

```suggestion
        (state: GlobalState, pluginId: string) => state.entities.admin.plugins?.[pluginId],
```

Reply:
Yeah it worked by adding `| undefined` into the result function